lib/propolis/src/block/mod.rs

```diff
@@ -244,7 +245,7 @@ pub trait Backend: Send + Sync + 'static {
     ///
     /// Spawning of any tasks required to do such request processing can be done
     /// as part of this start-up.
-    fn start(&self) -> anyhow::Result<()>;
+    fn start(&self) -> BoxFuture<'_, anyhow::Result<()>>;
```

Comment:
(We discussed this one a bit offline.)

I think most of the ergonomic problems in the implementations of this trait come from having callers who dynamically dispatch to it, which prevents the use of standard-issue `async fn`s in the trait. I revised this in aacaff8 to use the `async_trait` crate to at least get rid of some of the extra callee-side paperwork this created and filed #716 to track removing the Box allocation for non-async impls.

The other option is to convert the state driver back into an OS thread (as it was before this change), but as more async-heavy work moves into that context (e.g. driving live migration) so that it can call routines that call `block_on`, but I'm reluctant to go that way, because I anticipate more async-heavy work (like driving live migration) to move onto the state driver task over time.

Reply:
Yeah, moving back to an OS thread for pushing the state driver forward seems less than ideal.  With #716 filed, this works for me.